configuration/azure/appconfig/metadata.go

```diff
@@ -13,19 +13,63 @@ limitations under the License.
-
-	internalRequestTimeout        time.Duration `mapstructure:"-"`
-	internalMaxRetryDelay         time.Duration `mapstructure:"-"`
-	internalSubscribePollInterval time.Duration `mapstructure:"-"`
-	internalRetryDelay            time.Duration `mapstructure:"-"`
+	Host                  string        `mapstructure:"host"`
+	ConnectionString      string        `mapstructure:"connectionString"`
+	MaxRetries            int           `mapstructure:"maxRetries"`
+	MaxRetryDelay         time.Duration `mapstructure:"maxRetryDelay"`
+	RetryDelay            time.Duration `mapstructure:"retryDelay"`
+	SubscribePollInterval time.Duration `mapstructure:"subscribePollInterval"`
+	RequestTimeout        time.Duration `mapstructure:"requestTimeout"`
+}
+
+func (m *metadata) Parse(log logger.Logger, meta configuration.Metadata) error {
+	// Set defaults
+	m.MaxRetries = defaultMaxRetries
+	m.MaxRetryDelay = defaultMaxRetryDelay
+	m.RetryDelay = defaultRetryDelay
+	m.SubscribePollInterval = defaultSubscribePollInterval
+	m.RequestTimeout = defaultRequestTimeout
+
+	// Decode the metadata
+	decodeErr := contribMetadata.DecodeMetadata(meta.Properties, m)
+	if decodeErr != nil {
+		return decodeErr
+	}
+
+	// Validate options
+	if m.ConnectionString != "" && m.Host != "" {
+		return fmt.Errorf("azure appconfig error: can't set both %s and %s fields in metadata", host, connectionString)
+	}
+
+	if m.ConnectionString == "" && m.Host == "" {
+		return fmt.Errorf("azure appconfig error: specify %s or %s field in metadata", host, connectionString)
+	}
+
+	// In Dapr 1.11, these properties accepted nanoseconds as integers
+	// If users pass values larger than 10^6 (before 1ms; now 10^6 seconds), they probably set the metadata property for 1.11 in nanoseconds and that's not what they want here
+	// TODO: Remove this in Dapr 1.13
+	if m.MaxRetryDelay > time.Duration(10^6)*time.Second {
+		log.Warnf("[WARN] Property 'maxRetryDelay' is %v, which is probably incorrect. If you are upgrading from Dapr 1.11, please note that the property is now a Go duration rather than a number of nanoseconds", m.MaxRetryDelay)
+	}
+	if m.RetryDelay > time.Duration(10^6)*time.Second {
+		log.Warnf("[WARN] Property 'retryDelay' is %v, which is probably incorrect. If you are upgrading from Dapr 1.11, please note that the property is now a Go duration rather than a number of nanoseconds", m.RetryDelay)
+	}
+	if m.SubscribePollInterval > time.Duration(10^6)*time.Second {
+		log.Warnf("[WARN] Property 'subscribePollInterval' is %v, which is probably incorrect. If you are upgrading from Dapr 1.11, please note that the property is now a Go duration rather than a number of nanoseconds", m.SubscribePollInterval)
+	}
+	if m.RequestTimeout > time.Duration(10^6)*time.Second {
+		log.Warnf("[WARN] Property 'requestTimeout' is %v, which is probably incorrect. If you are upgrading from Dapr 1.11, please note that the property is now a Go duration rather than a number of nanoseconds", m.RequestTimeout)
+	}
```

Comment:
Thoughts on these @berndverst ? To make it better for users upgrading?

Feel free to remove if you don't like this

Reply:
@ItalyPaleAle Where does things like `m.RetryDelay` are getting set to passed in values from component.yaml?

Secondly, If I do not pass any value then these will be set to default values like - m.MaxRetryDelay will be `2m0s` which is greater than `time.Duration(10^6)*time.Second`. Which does not seem right.

logs - 
```
WARN[0000] [WARN] Property 'maxRetryDelay' is 2m0s, which is probably incorrect. If you are upgrading from Dapr 1.11, please note that the property is now a Go duration rather than a number of nanoseconds  app_id=test component="pravin-cilium-app-config (configuration.azure.appconfig/v1)" instance=Pravins-MacBook-Pro.local scope=dapr.contrib type=log ver=edge
```

Reply:
Duh, rookie mistake. `^` is a bitwise operator 😅 I'll fix it

> Where does things like m.RetryDelay are getting set to passed in values from component.yaml

When we call `DecodeMetadata` on the struct, they are populated. The method supports `time.Duration` automatically